Robot.cpp

```diff
@@ -0,0 +1,101 @@
+#include "Robot.h"
+
+Robot::Robot(double robotLength, double robotWidth, double wheelRadius, Camera & camera, double robotPosTol, double armPosTol) 
+    : robotLength(robotLength), robotWidth(robotWidth), wheelRadius(wheelRadius), camera(camera), robotPosTol(robotPosTol), armPosTol(armPosTol) {
+    // pid terms need to be determined experimentally
+    drive[frontLeft] = MotorController(10, 1, 1, 5, 0);
+    drive[backLeft] = MotorController(10, 1, 1, 5, 0);
+    drive[frontRight] = MotorController(10, 1, 1, 5, 0);
+    drive[backRight] = MotorController(10, 1, 1, 5, 0);
+
+    servoArm[x] = MotorController(10, 1, 1, 5, 0);
+    servoArm[y] = MotorController(10, 1, 1, 5, 0);
+    servoArm[z] = MotorController(10, 1, 1, 5, 0);
+
+    robotPosition = Point(0, 0, 0);
+    armPosition = Point(0, 0, 0);
+}
+
+Point Robot::getRobotPosition(){
+    return robotPosition;
+}
+
+Point Robot::getArmPosition(){
+    return armPosition;
+}
+
+// how are we going to do this? 
+// should it just be called at the end of each robot move function? 
```

Comment:
Not a change request. Do we want separate threads? Might be good for controllers to have more consistent timings.